layouts/docs/api-reference-links.html

```diff
@@ -0,0 +1,72 @@
+{{- define "main" -}}
+  <div class="td-content">
+    <h1>{{- .Title -}}</h1>
+
+    <!-- Existing concept page content goes here -->
+    {{- .Content -}}
+    
+    <!-- Define slice to store API reference links -->
+    {{- $apiReferenceLinks := slice -}}
+    {{- $apiReferenceText := T "api_reference_title" -}}
+    {{ $.Store.Delete "apiReferenceLinks" }}
+
+    <!-- Check if 'api-metadata' exists in the front-matter of the file -->
+    {{- if .Params.api_metadata -}}
+      <!-- Loop through each api_metadata entry -->
+      {{- range $metadata := .Params.api_metadata -}}
+        <!-- Extract API metadata -->
+        {{- $apiVersion := $metadata.apiVersion -}}
+        {{- $kind := $metadata.kind -}}
+        {{- $version := $apiVersion -}}
+        {{- $linkText := $metadata.override_link_text  | default $kind -}}
+
+       
+        <!-- Get all sections under the specified directory -->
+        {{- $apiRefBaseDir := "docs/reference/kubernetes-api/" -}}
+        {{- $apiRefSections := site.GetPage "section" $apiRefBaseDir -}}
+
+        <!-- Loop through sections -->
+        {{- range $apiRefSection := $apiRefSections.Sections -}}
+          {{- $apiRefDir := $apiRefSection.RelPermalink -}}
+          {{- $apiReferenceFiles := site.GetPage $apiRefDir -}}
+
+          <!-- Loop through API reference files -->
+          {{- range $apiRefFile := $apiReferenceFiles.RegularPages -}}
+            {{- $apiRefFileDirPath:= printf "/%s" $apiRefFile.File.Dir -}}
+
+            {{- if and (ne $apiRefFile.Section "") (in $apiRefFileDirPath $apiRefDir) -}}
+              <!-- Check if the file's metadata matches -->
+              {{- with $apiRefFile.Params.api_metadata -}}
+                {{- if and (eq .kind $kind) (eq .apiVersion $version) -}}
+                <!-- If the file's metadata matches, add link to variable -->
+                  {{- $link := printf "<a class='api-reference-page-link' href='%s'>%s %s</a>" $apiRefFile.Permalink $linkText $apiReferenceText -}}
+                  {{- $apiReferenceLinks = $apiReferenceLinks | append $link -}}
+                {{- end -}}
+              {{- end -}}
+            {{- end -}}
+          {{- end -}}
+        {{- end -}}
+
+      {{- end -}}
+    {{- end -}}
+
+      <!-- Display API reference heading and links if there are any -->
+      {{- if gt (len $apiReferenceLinks) 0 -}}
+      
+      <!-- Store the value of $apiReferenceLinks to display in meta link section in 'partials/page-meta-links.html' -->
+      {{ $.Store.Add "apiReferenceLinks" $apiReferenceLinks }}
+      
+      <h2 id="api-reference">{{- $apiReferenceText -}}</h2>
```

Comment:
A point worth mentioning is that while the new section containing [API reference links ](https://deploy-preview-45496--kubernetes-io-main-staging.netlify.app/docs/concepts/workloads/pods/#api-reference) will be added to the end of the page, the section title link is not included in the page's [Table of Contents](https://gohugo.io/methods/page/tableofcontents/). Similar to before, I can incorporate a javascript to append the newly added section to the Table of Contents, or alternatively, we could consider _omitting_ the addition of new section from the page altogether since we now have API reference links in the meta link section above the TOC.  I'm also open to any other suggestions or approaches.